chia/wallet/nft_wallet/nft_puzzle_utils.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import annotations
```

Comment:
I'm curious, what motivated the refactor into this new file?

Reply:
There was a circular import somewhere if the puzzles stayed in the same place as the utils

Reply:
I take it all of these functions were moved verbatim, right?

Reply:
Correct